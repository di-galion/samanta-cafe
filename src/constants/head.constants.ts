export const ORDER_ARRAY_OF_HEAD = ["Имя", "Телефон", "Улица", "Дом", "Подьезд", "Этаж","Комната", "Ключ", "Комментарий"]

export const PRODUCT_ARRAY_OF_HEAD = ["Название", "Описание", "Цена"]

export const CATEGORY_ARRAY_OF_HEAD = ["Название", "Slug"]