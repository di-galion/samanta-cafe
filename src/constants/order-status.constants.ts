export const NO_FILTER = 'No filtration'