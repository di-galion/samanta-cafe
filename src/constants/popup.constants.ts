export const POPUP_ID = "popup"
export const POPUP_ITEM_TEXT  = "Do you want to save edit?"