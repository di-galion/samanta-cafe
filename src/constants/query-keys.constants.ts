export const GET_ALL_ORDERS = "get all orders"
export const UPDATE_ORDER = "update order"
export const CREATE_ORDER = "create order"


export const CREATE_SAIL = "create sail"
export const UPDATE_SAIL = "update sail"

export const DELETE_SAIL = "delete sail"

export const CREATE_CATEGORY = "create category"
export const UPDATE_CATEGORY = "update category"
export const GET_ALL_CATEGORIES = 'get all categories'

export const GET_ALL_PRODUCTS = "get all products"
export const CREATE_PRODUCT = "create product"
export const UPDATE_PRODUCT = "update product"

export const GET_PROFILE = 'get profile'