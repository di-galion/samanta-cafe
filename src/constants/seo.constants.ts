export const SITE_NAME = "Samanta"