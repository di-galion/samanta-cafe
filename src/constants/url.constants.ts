export const ADMIN_URL = '/admin'
export const ORDER_URL = '/orders'
export const CATEGORY_URL = "/category"
export const PRODUCT_URL = "/product"

export const AUTH_URL = '/auth'

export const USER_URL = "/user"

export const DASHBOARD_URL = "/dashboard"