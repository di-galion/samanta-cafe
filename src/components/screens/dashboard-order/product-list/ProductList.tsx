"use client"
import styles from "./styles.module.scss"
import {useQuery} from "@tanstack/react-query";
import {OrderService} from "@/services/order/order.service";
import {ProductService} from "@/services/product/product.service";
import ProductListItem from "@/components/screens/dashboard-order/product-list-item/ProductListItem";
import {IProduct} from "@/types/product.types";
import {FC} from "react";
import {ISailCreate} from "@/types/sail.types";
import Loader from "@/components/ui/loader/Loader";
import {GET_ALL_PRODUCTS} from "@/constants/query-keys.constants";
import {useGetProducts} from "@/components/screens/dashboard-order/product-list/useGetProducts.hook";

export interface IProductList {
    setNewSailOnClick: (sail: ISailCreate) => void
    orderId: number
}
const ProductList: FC<IProductList> = ({setNewSailOnClick, orderId}) => {

    const {data, isLoading} = useGetProducts()

    return (
        <div className={styles.product_list_overlay}>
            <ul className={styles.product_list}>
                {!isLoading && data ?
                        data.map((product: IProduct)=> {
                            return (
                                <ProductListItem
                                    key={product.id}
                                    setNewSailOnClick={setNewSailOnClick}
                                    product={product}
                                    orderId={orderId}
                                />
                            )
                        }) :
                    <Loader />
                }
            </ul>
        </div>
    )
}

export default ProductList