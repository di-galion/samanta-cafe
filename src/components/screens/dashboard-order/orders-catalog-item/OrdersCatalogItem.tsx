"use client"
import {ChangeEvent, FC, useEffect, useRef, useState} from "react";
import { IOrderResponse} from "@/types/order.types";
import styles from './styles.module.scss'
import Select from "@/components/ui/select/Select";
import {EnumOrderStatus} from "@/components/screens/dashboard-order/orders-catalog-item/types/types";
import {useMutation, useQueryClient} from "@tanstack/react-query";
import {OrderService} from "@/services/order/order.service";
import FormData from "form-data";
import {ORDER_ARRAY_OF_HEAD} from "@/constants/head.constants";
import {getOrderHeadValues} from "@/utils/get-head-values.utils";
import SailItem from "@/components/screens/dashboard-order/sail-item/SailItem";
import Button from "@/components/ui/button/Button";
import ProductList from "@/components/screens/dashboard-order/product-list/ProductList";
import {ISail, ISailCreate} from "@/types/sail.types";
import {SailService} from "@/services/sail/sail.service";
import OptionsWrapper from "@/components/ui/admin/options-wrapper/OptionsWrapper";
import InputsSectionWrapper from "@/components/ui/admin/inputs-section-wrapper/InputsSectionWrapper";
import EditFormWrapper from "@/components/ui/admin/edit-form-wrapper/EditFormWrapper";
import Popup from "@/components/ui/popup/Popup";
import {POPUP_ITEM_TEXT} from "@/constants/popup.constants";
import {CREATE_SAIL, GET_ALL_ORDERS, UPDATE_ORDER} from "@/constants/query-keys.constants";
import {IProduct} from "@/types/product.types";
import {useUpdateOrderMutationHook} from "@/components/screens/dashboard-order/orders-catalog-item/useUpdateOrderMutation.hook";
import {
    useCreateSailMutation
} from "@/components/screens/dashboard-order/orders-catalog-item/useCreateSailMutation.hook";



const OrdersCatalogItem: FC<{order: IOrderResponse}> = ({order}) => {
    const [isEditing, setIsEditing] = useState(false)
    const [editItem, setEditItem] = useState({})
    const [isProductListOpen, setIsProductListOpen] = useState(false)
    const [newSail, setNewSail] = useState<ISailCreate>()
    const [isPopupOpen, setIsPopupOpen] = useState<boolean>(false)
    // Tp prevent use Effect during initial render
    const isMountedOrder = useRef(false);
    const isMountedSail = useRef(false);

    const form = useRef(null);

    const arrayOfValue = getOrderHeadValues(order)

    const mutate = useUpdateOrderMutationHook(order, editItem)
    const mutateSailCreate = useCreateSailMutation(order, newSail)
    useEffect(() => {
        if (isMountedOrder.current) mutate()
        else isMountedOrder.current = true
    }, [editItem])
    useEffect(() => {
        if (isMountedSail.current) mutateSailCreate()
        else isMountedSail.current = true

    }, [newSail])

    const onSubmit = (formElement: HTMLFormElement | null) => {

        if (!formElement) return;
        const form = new FormData(formElement as HTMLFormElement)

        if (form.get(ORDER_ARRAY_OF_HEAD[0]) === null) return
        setEditItem({
            name: form.get(ORDER_ARRAY_OF_HEAD[0]),
            tel: form.get(ORDER_ARRAY_OF_HEAD[1]),
            street: form.get(ORDER_ARRAY_OF_HEAD[2]),
            house: form.get(ORDER_ARRAY_OF_HEAD[3]),
            entrance: form.get(ORDER_ARRAY_OF_HEAD[4]),
            floor: form.get(ORDER_ARRAY_OF_HEAD[5]),
            room: form.get(ORDER_ARRAY_OF_HEAD[6]),
            key: form.get(ORDER_ARRAY_OF_HEAD[7]),
        })
        // if (Object.keys(editItem).length) mutate()
    }

    return (
        <EditFormWrapper >
            <form
                data-testid={"order-item-test"}
                ref={form}
                onSubmit={(e: ChangeEvent<HTMLFormElement>) => onSubmit(e)}
            >
                <OptionsWrapper
                    setIsEditing={setIsEditing}
                    isEditing={isEditing}
                    onConfirmClick={setIsPopupOpen}
                >
                    <Select
                        testId={"change-status-test"}
                        onClick={(value) => {
                            setEditItem({status: value})
                        }}
                        currentValue={order.status}
                    >
                        {Object.entries(EnumOrderStatus).map((value) => {
                            return <li data-testid={`select-item-${value[1]}-test`} key={value[1]}>{value[1]}</li>})}
                    </Select>
                </OptionsWrapper>

                <InputsSectionWrapper
                    isEditing={isEditing}
                    arrayOfValue={arrayOfValue}
                    arrayOfHead={ORDER_ARRAY_OF_HEAD}
                />

                <div className={styles.sails}>
                    {order.sails.map((sail) => {
                        return (<SailItem key={sail.id} sail={sail} />)
                    })}
                </div>

                {isProductListOpen &&
                    <ProductList
                        orderId={order.id}
                        setNewSailOnClick={(sail: ISailCreate) => {
                            setNewSail(sail)
                            if (newSail !== undefined) {
                                setIsProductListOpen(false)
                            }
                        }}
                    />
                }

                <div className={styles.button_add_sails}>
                    <Button
                        type={"button"}
                        testId={"button-add-sails-test"}
                        onClick={() => setIsProductListOpen(!isProductListOpen)}
                    >
                        Add sails
                    </Button>
                </div>

                <div className={styles.total}>{order.total}</div>

                {isPopupOpen &&
                    <Popup
                        content={POPUP_ITEM_TEXT}
                        onClickAgreeHandler={(e: ChangeEvent<HTMLFormElement>) => {
                            setIsEditing(false)
                            onSubmit(form.current)
                        }}
                        onClickDisagreeHandler={() => {
                            setIsEditing(false)
                            if (form.current) form.current.reset()
                        }}
                        setIsOpen={setIsPopupOpen}
                    />
                }
            </form>
        </EditFormWrapper>

    )
}
export default OrdersCatalogItem
