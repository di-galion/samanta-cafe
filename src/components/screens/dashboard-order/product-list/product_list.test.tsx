import {renderWithProviders} from "@/__tests__/helpers/renderTestApp";
import OrdersCatalogItem from "@/components/screens/dashboard-order/orders-catalog-item/OrdersCatalogItem";
import {ORDERS_TEST} from "@/components/screens/dashboard-order/orders-catalog/__tests__/constants";
import {screen} from "@testing-library/react";
import ProductList from "@/components/screens/dashboard-order/product-list/ProductList";
import {PRODUCTS_TEST} from "@/components/screens/product/__tests__/constants";
import {useGetProducts} from "@/components/screens/dashboard-order/product-list/useGetProducts.hook";


jest.mock("./useGetProducts.hook")
jest.spyOn({useGetProducts}, "useGetProducts").mockReturnValue({
    data: PRODUCTS_TEST,
    isLoading: false
})
describe("Product list", () => {

    afterEach(() => {
        jest.clearAllMocks()
    })

    test("Product list render", () => {
        jest.spyOn({useGetProducts}, "useGetProducts").mockReturnValue({
            data: PRODUCTS_TEST,
            isLoading: false
        })

        const component = renderWithProviders(<ProductList setNewSailOnClick={jest.fn()} orderId={1} />)

        expect(screen.getAllByTestId("product-list-item-test").length).toBe(PRODUCTS_TEST.length)
        expect(component).toMatchSnapshot()
    })

    test("Should render loader with isLoading = true", () => {
        jest.spyOn({useGetProducts}, "useGetProducts").mockReturnValue({
            data: PRODUCTS_TEST,
            isLoading: true
        })

        const component = renderWithProviders(<ProductList setNewSailOnClick={jest.fn()} orderId={1} />)

        expect(screen.queryByTestId("product-list-item-test")).toBeNull()
        expect(component.getAllByTestId("loader-test"))
        expect(component).toMatchSnapshot()
    })

})