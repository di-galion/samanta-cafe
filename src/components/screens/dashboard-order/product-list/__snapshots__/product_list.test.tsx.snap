// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list Product list render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product_list_overlay"
      >
        <ul
          class="product_list"
        >
          <li
            class="product_item"
            data-testid="product-list-item-test"
          >
            <img
              alt="picture"
              data-nimg="1"
              decoding="async"
              height="50"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2Flocal&w=128&q=75"
              srcset="/_next/image?url=http%3A%2F%2Flocal&w=64&q=75 1x, /_next/image?url=http%3A%2F%2Flocal&w=128&q=75 2x"
              style="color: transparent;"
              width="50"
            />
            <div
              class="info_container"
            >
              <div
                class="text"
              >
                f
              </div>
              <div
                class="description"
              >
                Lorem ipsum dolor sit.
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product_list_overlay"
    >
      <ul
        class="product_list"
      >
        <li
          class="product_item"
          data-testid="product-list-item-test"
        >
          <img
            alt="picture"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/_next/image?url=http%3A%2F%2Flocal&w=128&q=75"
            srcset="/_next/image?url=http%3A%2F%2Flocal&w=64&q=75 1x, /_next/image?url=http%3A%2F%2Flocal&w=128&q=75 2x"
            style="color: transparent;"
            width="50"
          />
          <div
            class="info_container"
          >
            <div
              class="text"
            >
              f
            </div>
            <div
              class="description"
            >
              Lorem ipsum dolor sit.
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Product list Should render loader with isLoading = true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product_list_overlay"
      >
        <ul
          class="product_list"
        >
          <div
            data-testid="loader-test"
            style="display: flex; justify-content: center; align-items: center;"
          >
            <svg
              aria-busy="true"
              aria-label="color-ring-loading"
              class=""
              data-testid="color-ring-svg"
              height="100"
              preserveAspectRatio="xMidYMid"
              role="status"
              viewBox="0 0 100 100"
              width="100"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs>
                <mask
                  id="ldio-4offds5dlws-mask"
                >
                  <circle
                    cx="50"
                    cy="50"
                    r="26"
                    stroke="#fff"
                    stroke-dasharray="122.52211349000194 40.840704496667314"
                    stroke-linecap="round"
                    stroke-width="9"
                    transform="rotate(198.018 50 50)"
                  >
                    <animatetransform
                      attributeName="transform"
                      dur="1s"
                      keyTimes="0;1"
                      repeatCount="indefinite"
                      type="rotate"
                      values="0 50 50;360 50 50"
                    />
                  </circle>
                </mask>
              </defs>
              <g
                mask="url(#ldio-4offds5dlws-mask)"
              >
                <rect
                  fill="#e15b64"
                  height="100"
                  width="15"
                  x="14.5"
                  y="0"
                >
                  <animate
                    attributeName="fill"
                    begin="-0.8s"
                    dur="1s"
                    keyTimes="0;0.25;0.5;0.75;1"
                    repeatCount="indefinite"
                    values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                  />
                </rect>
                <rect
                  fill="#f47e60"
                  height="100"
                  width="15"
                  x="28.5"
                  y="0"
                >
                  <animate
                    attributeName="fill"
                    begin="-0.6s"
                    dur="1s"
                    keyTimes="0;0.25;0.5;0.75;1"
                    repeatCount="indefinite"
                    values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                  />
                </rect>
                <rect
                  fill="#f8b26a"
                  height="100"
                  width="15"
                  x="42.5"
                  y="0"
                >
                  <animate
                    attributeName="fill"
                    begin="-0.4s"
                    dur="1s"
                    keyTimes="0;0.25;0.5;0.75;1"
                    repeatCount="indefinite"
                    values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                  />
                </rect>
                <rect
                  fill="#abbd81"
                  height="100"
                  width="15"
                  x="56.5"
                  y="0"
                >
                  <animate
                    attributeName="fill"
                    begin="-0.2s"
                    dur="1s"
                    keyTimes="0;0.25;0.5;0.75;1"
                    repeatCount="indefinite"
                    values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                  />
                </rect>
                <rect
                  fill="#849b87"
                  height="100"
                  width="15"
                  x="70.5"
                  y="0"
                >
                  <animate
                    attributeName="fill"
                    begin="0s"
                    dur="1s"
                    keyTimes="0;0.25;0.5;0.75;1"
                    repeatCount="indefinite"
                    values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                  />
                </rect>
              </g>
            </svg>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product_list_overlay"
    >
      <ul
        class="product_list"
      >
        <div
          data-testid="loader-test"
          style="display: flex; justify-content: center; align-items: center;"
        >
          <svg
            aria-busy="true"
            aria-label="color-ring-loading"
            class=""
            data-testid="color-ring-svg"
            height="100"
            preserveAspectRatio="xMidYMid"
            role="status"
            viewBox="0 0 100 100"
            width="100"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <defs>
              <mask
                id="ldio-4offds5dlws-mask"
              >
                <circle
                  cx="50"
                  cy="50"
                  r="26"
                  stroke="#fff"
                  stroke-dasharray="122.52211349000194 40.840704496667314"
                  stroke-linecap="round"
                  stroke-width="9"
                  transform="rotate(198.018 50 50)"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    keyTimes="0;1"
                    repeatCount="indefinite"
                    type="rotate"
                    values="0 50 50;360 50 50"
                  />
                </circle>
              </mask>
            </defs>
            <g
              mask="url(#ldio-4offds5dlws-mask)"
            >
              <rect
                fill="#e15b64"
                height="100"
                width="15"
                x="14.5"
                y="0"
              >
                <animate
                  attributeName="fill"
                  begin="-0.8s"
                  dur="1s"
                  keyTimes="0;0.25;0.5;0.75;1"
                  repeatCount="indefinite"
                  values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                />
              </rect>
              <rect
                fill="#f47e60"
                height="100"
                width="15"
                x="28.5"
                y="0"
              >
                <animate
                  attributeName="fill"
                  begin="-0.6s"
                  dur="1s"
                  keyTimes="0;0.25;0.5;0.75;1"
                  repeatCount="indefinite"
                  values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                />
              </rect>
              <rect
                fill="#f8b26a"
                height="100"
                width="15"
                x="42.5"
                y="0"
              >
                <animate
                  attributeName="fill"
                  begin="-0.4s"
                  dur="1s"
                  keyTimes="0;0.25;0.5;0.75;1"
                  repeatCount="indefinite"
                  values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                />
              </rect>
              <rect
                fill="#abbd81"
                height="100"
                width="15"
                x="56.5"
                y="0"
              >
                <animate
                  attributeName="fill"
                  begin="-0.2s"
                  dur="1s"
                  keyTimes="0;0.25;0.5;0.75;1"
                  repeatCount="indefinite"
                  values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                />
              </rect>
              <rect
                fill="#849b87"
                height="100"
                width="15"
                x="70.5"
                y="0"
              >
                <animate
                  attributeName="fill"
                  begin="0s"
                  dur="1s"
                  keyTimes="0;0.25;0.5;0.75;1"
                  repeatCount="indefinite"
                  values="#e15b64;#f47e60;#f8b26a;#abbd81;#849b87"
                />
              </rect>
            </g>
          </svg>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
